### PR TITLE
Default transformation

### PR DESCRIPTION
**Pull request by eladiw**
…lared fields in the anonymizer, which can be overrided by the user if supplied another default within the anonymizer template.

### CODE REVIEW

File: presidio-anonymizer/cmd/presidio-anonymizer/anonymizer/anonymizer.go [hunk 1]
```diff
@@ -11,6 +12,33 @@ import (
 
 type sortedResults []*types.AnalyzeResult
 
+// anonymizeSingleResult anonymize a single analyze result
+func anonymizeSingleResult(result *types.AnalyzeResult, transformations []*types.FieldTypeTransformation, text string) (bool, string, error) {
+	var err error
+	var newtext string
+	for _, transformation := range transformations {
+		if transformation.Fields == nil {
+			newtext, err = transformField(transformation.Transformation, result, text)
```
Comment (Ilanak) [inline presidio-anonymizer/cmd/presidio-anonymizer/anonymizer/anonymizer.go]: extract this 4 lines to a method - you're doing exactly the same here and in line 30

File: presidio-anonymizer/cmd/presidio-anonymizer/anonymizer/anonymizer.go [hunk 2]
```diff
@@ -35,33 +63,29 @@ func AnonymizeText(text string, results []*types.AnalyzeResult, template *types.
...
-			}
+		if transformed {
+			text = transformedText
+			continue
+		}
+
+		// Now, for any analyzer result which wasn't transformed, either
+		// transform using the default transformation, as described in the
+		// template, or fallback to default transformation
+		if template.DefaultTransformation != nil {
```
Comment (Ilanak) [inline presidio-anonymizer/cmd/presidio-anonymizer/anonymizer/anonymizer.go]: add test for the default transformation

File: presidio-anonymizer/cmd/presidio-anonymizer/anonymizer/anonymizer.go (hunk 1 above)
Comment (eladiw) [reply presidio-anonymizer/cmd/presidio-anonymizer/anonymizer/anonymizer.go]: 👍

File: presidio-anonymizer/cmd/presidio-anonymizer/anonymizer/anonymizer.go (hunk 2 above)
Comment (eladiw) [reply presidio-anonymizer/cmd/presidio-anonymizer/anonymizer/anonymizer.go]: 👍

**Comment by eladiw** [comment]
@Ilanak Done

**Comment by eladiw** [comment]
Note to reviewers, this PR won't pass the CI checks until the matching genproto PR won't be merged

**Pull request merged**